ubi_manifest/worker/tasks/depsolver/rpm_depsolver.py

```diff
@@ -52,6 +52,11 @@ def __init__(
         self._provides: set[RpmDependency] = set()  # set of rpm.provides we've visited
         self._requires: set[RpmDependency] = set()  # set of rpm.requires we've visited
 
+        self._required_files: set[str] = set()  # set of files required by visited RPMs
+        self._file_req_rpms: set[UbiUnit] = (
```

Comment:
Can you explain the usage of this? 
Do you think you could slightly re-design your approach so it's a bit closer to the approach used for `RpmDependency`?

I would suggest to use:
* ` self._required_files:` that's OK, but do only addition to this, do not remove items
*  add sth like  ` self._provided_files:` so you can easily check what's currently provided with current  RPMs
* then with quick set operations you can get for what we need to query/what's resolved etc.
* I also think you may need introduce the equivalent of `self._unsolved` for files  that A) will store remaining unresolved files B) it will be also use for stopping the `while` loop.
*  and then you can also easily log whether something wasn't resolved (which is currently not covered in this PR.)  - check usage of `self._log_warnings()`

Reply:
Oh, I see. So rather than store seen RPMs that have required files, just store which of the required files our current RPMs will cover. Then we can compare. That makes sense.